java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -2392,6 +2392,65 @@ void testWindowingRowNumber() {
     }
   }
 
+  @Test
+  void testWindowingCollect() {
+    Aggregation aggCollect = Aggregation.collect();
+    WindowOptions winOpts = WindowOptions.builder()
+                                         .minPeriods(1)
+                                         .window(2, 1).build();
+    StructType nestedType = new StructType(false,
+        new BasicType(false, DType.INT32), new BasicType(false, DType.STRING));
+    try (Table raw = new Table.TestBuilder()
+             .column( 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1) // GBY Key
+             .column( 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3) // GBY Key
+             .column( 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6) // OBY Key
```

Comment:
This is a little confusing. The `OBY` key has clashing values for the same value of `GBY` keys, indicating a non-deterministic ordering. It would be better to use distinct `OBY` values.

Reply:
> ~Edit: Please note that this is not wrong, just confusing.~

Sorry, no, I take that back. This is wrong. Please note line#2415:
```java
try (Table sorted = raw.orderBy(Table.asc(0), Table.asc(1), Table.asc(2)))
...
```
At this point, the `orderBy` cannot guarantee the expected order, since the values for all 3 sort keys are identical. Please switch the `OBY` column to use distinct keys (at least, for a specific combination of `GBY` keys). The following should work:
```java
    try (Table raw = new Table.TestBuilder()
             .column( 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1) // GBY Key
             .column( 1, 1, 1, 1, 2, 2, 2, 2, 3, 3, 3, 3) // GBY Key
             .column( 1, 2, 3, 4, 1, 2, 3, 4, 5, 6, 7, 8) // OBY Key
```

Reply:
I updated the OBY key as your suggestion.
I am a little confused that these 3 columns of GBY OBY keys are copied from other tests named testWindowingXXXX.
Could it be an issue in other windowing tests ?

Reply:
> Could it be either an issue in other windowing tests ?

You're right. I think this might be a problem with the other ones as well. If the values of the GBY/OBY columns all match across two rows, the relative order is indeterminate.

Thanks for making the change for this test. We can leave changing the other tests for a different time.